### PR TITLE
scorer docs overhaul

### PR DESCRIPTION
**Pull request by JRMeyer**

### CODE REVIEW

File: doc/Scorer.rst [hunk 1]
```diff
@@ -1,64 +1,83 @@
...
-=======================
+How to Train a Language Model
+=============================
 
-🐸STT pre-trained models include an external scorer. This document explains how to reproduce our external scorer, as well as adapt the scripts to create your own.
+Introduction
+------------
 
-The scorer is composed of two sub-components, a KenLM language model and a trie data structure containing all words in the vocabulary. In order to create the scorer package, first we must create a KenLM language model (using ``data/lm/generate_lm.py``, and then use ``generate_scorer_package`` to create the final package file including the trie data structure.
+This document explains how to train and format a language model for deployment.
```
Comment (reuben) [inline doc/Scorer.rst:9]: "format" a language model?

File: doc/Scorer.rst [hunk 2]
```diff
@@ -1,64 +1,83 @@
...
 
-The ``generate_scorer_package`` binary is part of the native client package that is included with official releases. You can find the appropriate archive for your platform in the `GitHub release downloads <https://github.com/coqui-ai/STT/releases/latest>`_. The native client package is named ``native_client.{arch}.{config}.{plat}.tar.xz``, where ``{arch}`` is the architecture the binary was built for, for example ``amd64`` or ``arm64``, ``config`` is the build configuration, which for building decoder packages does not matter, and ``{plat}`` is the platform the binary was built-for, for example ``linux`` or ``osx``. If you wanted to run the ``generate_scorer_package`` binary on a Linux desktop, you would download ``native_client.amd64.cpu.linux.tar.xz``.
+In practice you will almost always want to deploy a language model. A good language model will improve transcription accuracy by correcting predictable spelling and grammatical mistakes. If you can predict what kind of speech your 🐸STT will encounter, you can make great gains in accuracy with a custom language model.
```
Comment (reuben) [inline doc/Scorer.rst:11]: nit: Starting with "in practice" implies some previously discussed alternative. Feels weird to me. Plus the "almost". Like the alternative exists but it's not discussed anywhere.

File: doc/Scorer.rst (hunk 1 above)
Comment (reuben) [reply doc/Scorer.rst:9]: OK I see the usage below. I think "package" is a better description and matches the binary name.

**Review by reuben — APPROVED** [review]
You should also include this on the main toctree so it shows up on the sidebar.

File: doc/Scorer.rst (hunk 1 above)
Comment (JRMeyer) [reply doc/Scorer.rst:9]: for most people, there's no difference between the "scorer package" and the "language model", and the actual difference is only relevant to a small subset of people who will need more documentation than is on this page.

So, I think the framing of "training" and then "formatting" a language model is most useful. When I've explained this to people using the tools, I use this framing and it works better than anything else in my experience

File: doc/Scorer.rst (hunk 2 above)
Comment (JRMeyer) [reply doc/Scorer.rst:11]: I discuss this later down, but I see your point, esp. about "in practice"

File: doc/Scorer.rst (hunk 1 above)
Comment (reuben) [reply doc/Scorer.rst:9]: Sorry, I meant using the *verb* to package, ie. "how to train and package a language model for deployment". I think that still captures your intent but matches the tool name, no?

Comment (JRMeyer) [reply doc/Scorer.rst:9]: ah yes, that's perfect:)

**Pull request merged**